src/server.h

```diff
@@ -1197,6 +1209,11 @@ typedef struct client {
     int multibulklen;    /* Number of multi bulk arguments left to read. */
     long bulklen;        /* Length of bulk argument in multi bulk request. */
     long long woff;      /* Last write global replication offset. */
+    /* Parsed commands queue */
+    commandParserState *cmd_queue; /* Queue of parsed commands. */
+    short cmd_queue_len;           /* Number of elements in the queue. */
+    short cmd_queue_off;           /* Offset to the next element to execute. */
+    short cmd_queue_cap;           /* Allocation size (capacity) of the cmd_queue array. */
```

Comment:
Consider putting it all in one struct cmdQueue

Reply:
I considered it. Also considered implementing some generic queue, but the code got simpler without too much abstraction.

Do you mean to put the queue struct inline in the client struct or to allocate it and store a pointer to it in the client struct?